heka/files/toml/global.toml

```diff
@@ -4,7 +4,7 @@
 maxprocs={{ workers }}
 base_dir="/var/cache/{{ service_name }}"
 
-hostname="{{ grains.fqdn.split('.')[0] }}"
+hostname="{{ pillar.linux.system.name|default(grains.host) }}"
```

Comment:
I think the default filter won't work if pillar.linux is completely undefined.
But following should work:
```pillar.get('linux',{}).get('system',{}).name|default(grains.host```